### PR TITLE
Fix #1385: Temporarily lift 22 limit for functions

### LINKED ISSUE #1385: Lambdas with parametrized captured parameters may end with more than 22 parameters.
```scala
 def foo22[T](c1:T, c2:T, c3:T, c4:T, c5:T, c6:T, c7:T, c8:T, c9:T, c10:T, c11:T, c12:T, c13:T, c14:T, c15:T, c16:T, c17:T, c18:T, c19:T, c20:T, c21:T, c22:T): Int => T = {
    (a: Int) => { // This labda ends with 23 parameters (c1 to c22 and a)
      c22; c21; c20; c19; c18; c17; c16; c15; c14; c13; c12; c11; c10; c9; c8; c7; c6; c5; c4; c3; c2; c1
    }
  }
```

This case will fail because the inner lambda will have 23 parameter, which fails due to #1384.

### PR DESCRIPTION
**Pull request by odersky**
Explanatiions in comment in Definitions. Review by @nicolasstucki.

### CODE REVIEW

File: src/dotty/tools/dotc/core/Definitions.scala [hunk 1]
```diff
@@ -12,7 +12,13 @@ import collection.mutable
 import scala.reflect.api.{ Universe => ApiUniverse }
 
 object Definitions {
-  val MaxFunctionArity, MaxTupleArity = 22
+  val MaxTupleArity, MaxAbstractFunctionArity = 22
+  val MaxFunctionArity = 24
```
Comment (nicolasstucki) [inline src/dotty/tools/dotc/core/Definitions.scala]: `MaxFunctionArity` should be 44 as documented below.

Comment (nicolasstucki) [inline src/dotty/tools/dotc/core/Definitions.scala]: Actually it is the doc that is off.

**Comment by nicolasstucki** [comment]
That is all.

File: tests/pos/i1385.scala [hunk 2]
```diff
@@ -0,0 +1,7 @@
+object Test {
+def foo22[T](c1:T, c2:T, c3:T, c4:T, c5:T, c6:T, c7:T, c8:T, c9:T, c10:T, c11:T, c12:T, c13:T, c14:T, c15:T, c16:T, c17:T, c18:T, c19:T, c20:T, c21:T, c22:T): Int => T = {
+    (a: Int) => { // This labda ends with 23 parameters (c1 to c22 and a)
```
Comment (xeno-by) [inline tests/pos/i1385.scala]: Typo

**Comment by nicolasstucki** [comment]
LGTM

**Pull request merged**